ping-payload-extensions/ping-custom-payload-extensions.md

```diff
@@ -0,0 +1,110 @@
+# Ping Custom Payload Extensions
+
+## The problem
+
+Over time we will need to change what is sent over ping/pong `custom_payload`'s to adjust for what is needed in the future.
+Currently the only way to update a `custom_payload` is through breaking changes which break protocol compatible between Portal implementations.
+As we get users, deploying breaking changes to mainnet will no longer be an option as we are aiming for a 100% uptime.
+
+## The Solution
+
+Ping Payload Extensions. There is a minimal set of `standard extensions` which require a fork hard fork to change. This framework allows Portal clients to implement `non standard extensions` which don't require a hard fork to deploy to the network. A more flexible way to extend the Protocol without bloating the [Portal-Wire-Protocol](../portal-wire-protocol.md)
```

Comment:
I'm not sure that the "problem/solution" belongs in the spec.  Once this is merged, the "problem" part is no longer applicable and I think feels awkward.

I would remove the "problem" part and change "the solution" part to be "motivation" and include the parts from the problem statement about needing a way to communicate meta data without modifying the base protocol.